src/camera.py

```diff
@@ -0,0 +1,276 @@
+        """
+        Tilts the camera.
+
+        Controls the vertical movement of the camera in [0, 150] degrees.
+        0 is directly in front of the car, 90 faces upwards, and 150 faces the
+        Raspberry Pi. The degrees will be clamped to [0, 150].
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        if self._camera_type == ABSOLUTE:
+            self._current_tilt_angle = np.clip(degree, self.MIN_PAN_TILT_DEGREE,
+                                                       self.MAX_TILT_DEGREE)
+        else:
+            self._current_tilt_angle = np.clip(degree + self._current_tilt_angle,
+                                                        self.MIN_PAN_TILT_DEGREE,
+                                                        self.MAX_TILT_DEGREE)
+
+        self._tilt_servo.write(np.clip(self._current_tilt_angle
+                                       + self._tilt_offset,
+                                       self.MIN_PAN_TILT_DEGREE,
+                                       self.MAX_TILT_DEGREE))
+
+
+    def calibrate_tilt(self, degree):
+        """
+        Calibrates the camera's tilting servo.
+
+        Writes a new tilt offset value to the config file and changes the current
+        tilt offset. The calibration value can be negative.
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        if self._camera_type == ABSOLUTE:
+            self._tilt_offset = int(np.clip(degree, self.MIN_TILT_CALI_DEGREE,
+                                                    self.MAX_TILT_CALI_DEGREE))
+        else:
+            self._tilt_offset = int(np.clip(degree + self._tilt_offset,
+                                                     self.MIN_TILT_CALI_DEGREE,
+                                                     self.MAX_TILT_CALI_DEGREE))
+
+        self._db.set('tilt_offset', self._tilt_offset)
+
+
+    def get_tilt_angle(self):
+        """
+        Gets the current tilt angle of the camera.
+
+        :return: The current tilt angle (in degrees).
+        :rtype: int
+        """
+        return self._current_tilt_angle
+
+
+    def reset_camera(self):
+        """
+        Resets the camera to default position.
+
+        Sets the camera pan to 90 and the tilt to 0.
+        """
+        self._current_pan_angle = 90
+        self._current_tilt_angle = 0
+        self._pan_servo.write(self._current_pan_angle)
+        self._tilt_servo.write(self._current_tilt_angle)
+
+
+def main():
+    """
+    Demonstrates the camera's panning and tilting functionality.
+
+    Creates a camera object and pans from 0 to 180, then from 180 to 0. The
+    camera then tilts from 0 to 150, then 150 to 0.
+    """
+    camera_delay = 0.05
+    picar.setup()
+
+    # Creates a Camera object with absolute degrees (default).
+    camera = Camera()
+    print "Testing absolute camera degrees."
+
+    # Moving the camera to the far right for panning.
+    camera.pan(0)
+
```

Comment:
consistency: you should sleep after this line because you sleep on line 230.

Reply:
agreed. whoops.